### DISCUSSION TITLE
Set tags automatically based on rules

### DISCUSSION THREAD

**Discussion by SansGuidon** [shaarli/Shaarli#2095] [Ideas]
I'm using both Wallabag and Shaarli but the benefit of Wallabag over Shaarli is that we can configure it to assign specific tags for links based on keywords in title, url or content, see https://github.com/wallabag/doc/blob/master/en/user/configuration/tagging_rules.md

I wonder if Shaarli could support this by design or via a plugin.
is there anyone else feeling this is needed ?

**Comment by SansGuidon** [comment]
I think this could be a plugin but I think the tags attribute is not editable any longer since v0.12.x, which IMHO reveals issues with Shaarli version numbers with respect to semantic versioning practices, as any breaking change should trigger a major version bump and because welcoming plugins is maybe no longer a priority, many plugins are broken and it's harder for users to find good ones, find reliable shaarli docs and make new plugins, so it could be harder for users to keep using Shaarli.

@nodiscc any idea if it's doable through a plugin? I've asked a similar question in https://github.com/shaarli/Shaarli/issues/2000#issuecomment-2474417392 so feel free to answer wherever :-)

Thanks!

**Comment by kcaran** [comment]
Maybe I'm misunderstanding the issue, but you definitely can add or edit tags when you add a bookmark. I have a youtube plugin that automatically adds tags from the youtube metadata:

https://github.com/kcaran/shaarli-plugin-youtube

**Comment by SansGuidon** [comment]
Thank you @kcaran , I've spent half a day yesterday trying to update tags through code and not succeeding, I mean only editing the title and description would work but never the tags. Then while troubleshooting I accidentally corrupted my shaarli db. Then I had to spend another few couple hours making my Shaarli work again.
Too bad I didn't find your plugin earlier in examples.

**Reply by nodiscc** [reply]
> accidentally corrupted my shaarli db

I highly suggest making regular backups of the datastore before attempting to touch code that writes to it.

As you guessed it is still possible to edit tags through a plugin. #2000 is unrelated (see comments there)

@kcaran would you like to add https://github.com/kcaran/shaarli-plugin-youtube to https://github.com/shaarli/Shaarli/blob/master/doc/md/Community-and-related-software.md (and other plugins you may have written?)

**Reply by SansGuidon** [reply]
Thanks, I could save a lot of time now by basing my plugin on the youtube plugin :-)
And I believe I could make my work public also, after improving the UX of the plugin.
I also support the idea to add the youtube plugin to the list of recommended plugins, as well as any possible plugin still being developed, because that's part of attracting users to Shaarli.

**Comment by SansGuidon** [comment]
For future reference, in case someone is interested in implementing a similar feature through the plugin system, here is my working POC, https://gitea.zoemp.be/sansguidon/shaarli_plugins/src/branch/main/auto_tag_plugin

**Reply by nodiscc** [reply]
Thanks for keeping this discussion updated @SansGuidon 
Feel free to send a pull request to update https://github.com/shaarli/Shaarli/blob/master/doc/md/Community-and-related-software.md with your plugins once they are well tested (a bit more documentation would be expected though)
